### PR TITLE
Clarify hooks

### PR DESCRIPTION
**Pull request by plaindocs**
Fix for #758

### CODE REVIEW

File: pages/agent/v3/hooks.md.erb [hunk 1]
```diff
@@ -66,44 +88,26 @@ The following is a complete list of hooks that can be implemented, and the order
 
 ## Creating hook scripts
 
-Hooks are bash scripts you can use to execute commands and export environment variables. All the standard [Buildkite environment variables](/docs/pipelines/environment-variables) are available to be used inside your hook scripts.
+Hooks are shell scripts you can use to run commands and export environment variables.
+Hooks have access to all the standard [Buildkite environment variables](/docs/pipelines/environment-variables).
 
-The following is an example of a custom `environment` hook which exports a GitHub API key for the pipeline’s release build step:
+Hooks are copied to `$TMPDIR` directory and run by the default shell on the Agent that is running the build, which has a few implications:
```
Comment (DWSR) [inline pages/agent/v3/hooks.md.erb:94]:
```suggestion
Hooks are copied to `$TMPDIR` directory and are **sourced by** the default shell on the Agent that is running the build, which has a few implications:
```

File: pages/agent/v3/hooks.md.erb [hunk 2]
```diff
@@ -66,44 +88,26 @@ The following is a complete list of hooks that can be implemented, and the order
...
 
-```bash
-set -eu
-echo '--- \:house_with_garden\: Setting up the environment'
+* `$BASH_SOURCE` contains the location where the hook is stored
```
Comment (DWSR) [inline pages/agent/v3/hooks.md.erb:96]:
```suggestion
* `$BASH_SOURCE` contains the location where the hook script was originally located
```

File: pages/agent/v3/hooks.md.erb [hunk 3]
```diff
@@ -66,44 +88,26 @@ The following is a complete list of hooks that can be implemented, and the order
...
+* `$0` contains the location of the copy of the script that is running
```
Comment (DWSR) [inline pages/agent/v3/hooks.md.erb:97]:
```suggestion
* `$0` contains the location of the copy of the script that is running from `$TMPDIR`
```

File: pages/agent/v3/hooks.md.erb [hunk 4]
```diff
@@ -1,24 +1,46 @@
...
+
+### Repository hooks
+
+Repository hooks allow you to execute repository-specific scripts. Repository hooks live alongside your repository’s source code under the directory `.buildkite/hooks`.
+
+To get started create a `.buildkite/hooks` directory, add your hook script, and make it executable.
+
+### Plugin hooks
+
+Plugin hooks allow any plugins you’ve defined to execute and override any of the default behavior.
```
Comment (DWSR) [inline pages/agent/v3/hooks.md.erb:29]:
```suggestion
Plugin hooks allow any plugins you’ve defined as part of the pipeline step to override any of the default behavior.
```

Comment (plaindocs) [reply pages/agent/v3/hooks.md.erb:29]: This is a good call. We don't currently lint for that because 

```
$ ack -ch execute*
819
```

**Comment by plaindocs** [comment]
Cheers @DWSR - that is all super useful! <3

**Pull request merged**